Evolutionary_loop/Loop_Parts/Part_B/Part_B_PUEO_job2.sh

```diff
@@ -0,0 +1,116 @@
+#!/bin/bash
+
+########    XF Simulation Software (B)     ########################################################################################## 
+#
+#
+#     1. Prepares output.xmacro with generic parameters such as :: 
+#             I. Antenna type
+#             II. Population number
+#             III. Grid size
+#
+#
+#     2. Prepares simulation_PEC.xmacro with information such as:
+#             I. Each generation antenna parameters
+#
+#
+#     3. Runs XF and loads XF with both xmacros. 
+#
+#
+###################################################################################################################################### 
+# varaibles
+indiv=$1
+gen=$2
+NPOP=$3
+WorkingDir=$4
+RunName=$5
+XmacrosDir=$6
+XFProj=$7
+GeoFactor=$8
+num_keys=$9
+NSECTIONS=${10}
+XFCOUNT=${11}
+
+#chmod -R 777 /fs/ess/PAS1960/BiconeEvolutionOSC/BiconeEvolution/
+
+## Lines for output.xmacro files ##
+## I've commented these out because we needed to put them inside of a loop in the macroskeleton ##
+## Currently these are hardcoded outputmacroskeleton_GPU.xmacro ##
+#line1='var query = new ResultQuery();'
+#line2='///////////////////////Get Theta and Phi Gain///////////////'
+#line3='query.projectId = App.getActiveProject().getProjectDirectory();'
+freqlist="20000 21000 22000 23000 24000 25000 26000 27000 28000 29000 30000 31000 32000 33000 34000 35000 36000 37000 38000 39000 40000 41000 42000 43000 44000 45000 46000 47000 48000 49000 50000 51000 52000 53000 54000 55000 56000 57000 58000 59000 60000 61000 62000 63000 64000 65000 66000 67000 68000 69000 70000 71000 72000 73000 74000 75000 76000 77000 78000 79000 80000 81000 82000 83000 84000 85000 86000 87000 88000 89000 90000 91000 92000 93000 94000 95000 96000 97000 98000 99000 100000 101000 102000 103000 104000 105000 106000 107000 108000 109000 110000 111000 112000 113000 114000 115000 116000 117000 118000 119000 120000 121000 122000 123000 124000 125000 126000 127000 128000 129000 130000 131000 132000 133000 134000 135000 136000 137000 138000 139000 140000 141000 142000 143000 144000 145000 146000 147000 148000 149000 150000"
+#The list of frequencies, scaled up by 100 to avoid float operation errors in bash
+#we have to wait to change the frequencies since we're going to be changing them as we append them to simulation_PEC.xmacro (which is removed below before being remade)
+
+## We need the loop to pause until all the XF jobs are done
+## To do this, we'll just count the flag files
+
+cd $WorkingDir/Run_Outputs/$RunName/GPUFlags/
+flag_files=$(ls | wc -l) #$(ls -l --file-type | grep -v '/$' | wc -l)
```

Comment:
```suggestion
flag_files=0
```
If you expect this to be always `< $XFCOUNT`, you can set it to `0` to avoid some duplication.  This means that it will directly wait for 1m before checking if there are enough files.

Reply:
Got it, thanks. 
But I think I should be editing this in the PUEO repo instead of this one.
Or maybe I could somehow use a submodule for PUEO scripts?

Reply:
Not sure about where it would be better to fix it, but regardless, there should be as little duplication as possible -- ideally none at all.

Reply:
Got it, thanks!
I'll look into how to bring in individual files using something analogous to git submodule.
Regarding avoiding duplication, this is what I am currently working on -- to merge multiple old scripts into one.
Since I didn't write them though, I am not too familiar with them, so I try to delete as little as possible when editing.

Reply:
Makes sense.
At some point I wanted to discuss with the team pros and cons of having a mix of bash/Python/C++, and whether it would make sense to move some of these bash scripts to Python.

Reply:
Actually I was thinking changing some of the python scripts into C++ for consistency, since the GA (called in Part A) is probably inevitably going to stay in C++ at this point.

Reply:
I brought this up on Slack.  Let's discuss there or in one a meeting with the rest of the team.